src/components/shared/ContextMenu.tsx

```diff
@@ -1,64 +1,58 @@
-  children: React.ReactNode;
-  menuId: string;
-  value: any;
+  outerRef: any;
+  addRowHandler: (row: number) => void;
+  deleteRowHandler: (row: number) => void;
 }
 
-interface MenuItemProps {
-  event: React.TouchEvent<HTMLDivElement> | React.MouseEvent<HTMLDivElement, MouseEvent>;
-  target: HTMLElement;
-}
+export const ContextMenu = ({
+  outerRef,
+  addRowHandler,
+  deleteRowHandler,
+}: ContextMenuProps) => {
+  const { xPos, yPos, menu, cellValue, cellRow } = useContextMenu(outerRef);
+  const classes = useStyles();
 
-interface FooObject {
-  foo: string;
-}
-
-//This doesn't come with default styling unfortunately. They're style sheets are here: https://github.com/vkbansal/react-contextmenu/blob/master/examples/react-contextmenu.css
-export const ContextMenu = ({ children, menuId, value }: ContextMenuProps) => {
-  function handleClick({ target }: MenuItemProps, data: FooObject) {
-    console.log(data, target.innerText);
-  }
-
-  const CellMenu = (
-    <div>
-      <ContextMenuComponent
-        id={menuId}
-        hideOnLeave={true}
-        style={{
-          margin: 1,
-          borderColor: 'black', //these border attributes don't show up for some reason
-          borderWidth: 1,
-          padding: 5,
-          backgroundColor: 'white',
-          opacity: 1,
-          zIndex: 10,
-        }}
+  const rowIndex = parseInt(cellRow);
+  if (menu) {
+    return (
+      <ul
+        className={classes.menu}
+        style={{ position: 'fixed', top: yPos, left: xPos, zIndex: 100 }}
       >
-        <MenuItem data={{ foo: 'bar' }} onClick={handleClick}>
-          {value}
-        </MenuItem>
-        <MenuItem data={{ foo: 'bar' }} onClick={handleClick}>
-          ContextMenu Item 2
-        </MenuItem>
-        <MenuItem divider />
-        <MenuItem data={{ foo: 'bar' }} onClick={handleClick}>
-          ContextMenu Item 3
-        </MenuItem>
-      </ContextMenuComponent>
-    </div>
-  );
-
-  return (
-    <>
-      <ContextMenuTrigger id={menuId}>{children}</ContextMenuTrigger>
-      {CellMenu}
-    </>
-  );
+        <li className={classes.menuItem}>{cellValue}</li>
+        <li className={classes.menuItem} onClick={() => addRowHandler(rowIndex)}>
+          Add new row above
+        </li>
+        <li className={classes.menuItem} onClick={() => addRowHandler(rowIndex + 1)}>
+          Add new row below
+        </li>
+        <li className={classes.menuItem} onClick={() => deleteRowHandler(rowIndex)}>
+          Delete row
+        </li>
+      </ul>
+    );
+  }
+  return <></>;
 };
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    menu: {
+      borderWidth: 1,
+      borderColor: 'black',
+      borderStyle: 'solid',
+    },
+    menuItem: {
+      listStyleType: 'none',
+      borderWidth: 1,
+      borderColor: 'black',
+      borderStyle: 'solid',
+      backgroundColor: 'white',
+    }, //would like hover and click styling, i don't remember how to do that stuff
```

Comment:
for hover and click styling, you can do something like this:
```js
    menuItem: {
      listStyleType: 'none',
      borderWidth: 1,
      borderColor: 'black',
      borderStyle: 'solid',
      backgroundColor: 'white',
      '&:hover, &:focus, &:active': {
            color: 'black',
            background: 'blue',
      }
    }, //would like hover and click styling, i don't remember how to do that stuff
```
`:hover` for hover, `:focus` for when you tabbing focus to it, `:active` for when you click on it

Reply:
looks good, thanks